File: c++/sort-linked-list-of-0s-1s-2s.cpp
// https://practice.geeksforgeeks.org/problems/given-a-linked-list-of-0s-1s-and-2s-sort-it/1
Node* segregate(Node *head) {
     // Create couters of 0, 1 and 2
    if(head == NULL)
        return head;
    int count[] = {0, 0, 0};
    Node *current = head;
    while(current != NULL)
    {
        count[current->data] += 1;
        current  = current->next;
    }
    int index = 0;
    // GO to head of the linked list
    current = head;
    while(current != NULL)
    {
        while(count[index] > 0)
        {
            count[index] -=1;
            current->data = index;
            current = current->next;
        }
        index++;
    }
    return head; 
}

File: python/sort-linked-list-of-0s-1s-2s.py
"""
https://practice.geeksforgeeks.org/problems/given-a-linked-list-of-0s-1s-and-2s-sort-it/1?page=1&category[]=Linked%20List&curated[]=1&sortBy=submissions
Given a linked list of N nodes where nodes can contain values 0s, 1s, and 2s only. The task is to segregate 0s, 1s, and 2s linked list such that all zeros segregate to head side, 2s at the end of the linked list, and 1s in the mid of 0s and 2s.
"""

class Solution:
    #Function to sort a linked list of 0s, 1s and 2s.
    def segregate(self, head):
        copy_head_start = head
        number=""
        while copy_head_start:
            number += (str(copy_head_start.data))
            copy_head_start = copy_head_start.next
        number = sorted(number)
        copy_head_start = head
        for digit in number:
            copy_head_start.data = int(digit)
            copy_head_start = copy_head_start.next
        return head